Frameworks/AEPSwiftUI/Sources/ContentCards/CardDelegate.swift

```diff
@@ -0,0 +1,27 @@
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+/// CardDelegate protocol defines methods that a delegate should implement to handle various card related events from the User Interface
+protocol CardDelegate {
```

Comment:
I was reserving the protocol name ContentCardEventListening for the public API's... And that why I thought i would name this differently (and took the inspiration from UIKit ViewController approach.  (datasource and delegate)).

I am trying to refactor it so that I won't need this protocol. For now I will remove it and then add them if needed

Reply:
I was reserving the protocol name ContentCardEventListening for the public API's... And that why I thought i would name this differently (and took the inspiration from UIKit ViewController approach.  (datasource and delegate)).

I am trying to refactor it so that I won't need this protocol. For now, I will remove it and then add them if needed